blockchain/validators/src/metadata/avr/config.rs

```diff
@@ -0,0 +1,153 @@
+    ///
+    /// # Arguments
+    ///
+    /// * `path` - Path to the configuration file containing
+    /// the history of AVRs generated by MobileCoin consensus node
+    /// enclaves
+    pub fn load(path: impl AsRef<Path>) -> Result<AvrHistoryConfig, ParseError> {
+        let data = fs::read_to_string(path)?;
+
+        if let Ok(mut config) = serde_json::from_str(&data): Result<AvrHistoryConfig, _> {
+            config.node.sort();
+            Ok(config)
+        } else if let Ok(mut config) = toml::from_str(&data): Result<AvrHistoryConfig, _> {
+            config.node.sort();
+            Ok(config)
+        } else {
+            Err(ParseError::UnsuportedFileFormat)
+        }
+    }
+}
+
+#[serde_as]
+#[derive(Deserialize, Serialize)]
+#[serde(remote = "VerificationReport")]
+/// Struct to shadow the mc_blockchain_types's VerificationReport for
+/// serialization purposes
+pub struct VerificationReportShadow {
+    /// Report Signature bytes, from the X-IASReport-Signature HTTP header.
+    #[serde_as(as = "hex::Hex")]
+    pub sig: VerificationSignature,
+
+    /// Attestation Report Signing Certificate Chain, as an array of
+    /// DER-formatted bytes, from the X-IASReport-Signing-Certificate HTTP
+    /// header.
+    #[serde_as(as = "Vec<hex::Hex>")]
+    pub chain: Vec<Vec<u8>>,
+
+    /// The raw report body JSON, as a byte sequence
+    pub http_body: String,
+}
+
+// SerializeAs and Deserialize are needed to get VerificationReportShadow (serde
+// remote) to work with container types (ie. Option<VerificationReport> )
+impl SerializeAs<VerificationReport> for VerificationReportShadow {
+    fn serialize_as<S>(source: &VerificationReport, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        VerificationReportShadow::serialize(source, serializer)
+    }
+}
+
+impl<'de> DeserializeAs<'de, VerificationReport> for VerificationReportShadow {
+    fn deserialize_as<D>(deserializer: D) -> Result<VerificationReport, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        VerificationReportShadow::deserialize(deserializer)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::test_utils::{
+        get_avr_history_config, SAMPLE_AVR_HISTORY_JSON, SAMPLE_AVR_HISTORY_TOML,
+    };
+    use serde_json;
+    use std::fs;
+    use tempfile::TempDir;
+
+    #[test]
+    fn test_avr_history_serialization_roundtrip_works() {
+        // Get a manually constructed AVR history config to act as control data
+
+        let control_avr_history = get_avr_history_config();
+
+        // Serialize the config to JSON & TOML
+        let toml_str = toml::to_string_pretty(&control_avr_history).unwrap();
+        let json_str = serde_json::to_string_pretty(&control_avr_history).unwrap();
+
+        let history_from_toml: AvrHistoryConfig = toml::from_str(toml_str.as_str()).unwrap();
+        let history_from_json: AvrHistoryConfig = serde_json::from_str(json_str.as_str()).unwrap();
```

Comment:
nit: can use `&` instead of `as_str()`
```suggestion
        let history_from_toml: AvrHistoryConfig = toml::from_str(&toml_str).unwrap();
        let history_from_json: AvrHistoryConfig = serde_json::from_str(&json_str).unwrap();
```